tgui/packages/tgui/components/TextArea.js

```diff
@@ -104,24 +130,19 @@ export class TextArea extends Component {
     if (input) {
       input.value = toInputValue(nextValue);
     }
-
     if (this.props.autoFocus || this.props.autoSelect) {
-      setTimeout(() => {
-        input.focus();
-
-        if (this.props.autoSelect) {
-          input.select();
-        }
-      }, 1);
+      Byond.subscribeTo('open', () => {
+        this.getFocus();
+      });
+      this.getFocus();
```

Comment:
Please revert this, first of all subscribeTo is going to be a huge memory leak in a commonly used component like this (because unsubscribe is not implemented), second of all tgui host does not implement the "open" message, so this is completely specific to your tgui-modal.

Instead implement focusing logic yourself in tgui-modal. Implement an `innerRef` property that you could pass to this component (would initialize `this.textareaRef` if innerRef property is defined), this would allow getting a reference to textarea and doing whatever you want to it tgui-modal.